internal/lsp/server.go

```diff
@@ -447,6 +450,59 @@ func (l *LanguageServer) StartCommandWorker(ctx context.Context) {
 					commands.ParseOptions{TargetArgIndex: 0, RowArgIndex: 1, ColArgIndex: 2},
 					params,
 				)
+			case "regal.eval":
+				fmt.Fprintf(os.Stderr, "regal.eval called with params: %v\n", params)
+
+				if len(params.Arguments) != 3 {
+					l.logError(fmt.Errorf("expected three arguments, got %d", len(params.Arguments)))
+
+					break
+				}
+
+				file, ok := params.Arguments[0].(string)
+				if !ok {
+					l.logError(fmt.Errorf("expected first argument to be a string, got %T", params.Arguments[0]))
+
+					break
+				}
+
+				path, ok := params.Arguments[1].(string)
+				if !ok {
+					l.logError(fmt.Errorf("expected second argument to be a string, got %T", params.Arguments[1]))
+
+					break
+				}
+
+				line, ok := params.Arguments[2].(float64)
+				if !ok {
+					l.logError(fmt.Errorf("expected third argument to be a number, got %T", params.Arguments[2]))
+
+					break
+				}
+
+				input := FindInput(
+					uri.ToPath(l.clientIdentifier, file),
+					uri.ToPath(l.clientIdentifier, l.workspaceRootURI),
+				)
+
+				result, err := l.EvalWorkspacePath(ctx, path, input)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "failed to evaluate workspace path: %v\n", err)
+
+					break
+				}
+
+				responseParams := map[string]any{
+					"result": result,
+					"line":   line,
+				}
+
+				responseResult := map[string]any{}
+
+				err = l.conn.Call(ctx, "regal/showEvalResult", responseParams, &responseResult)
```

Comment:
We must use a custom event here as there is client code that will actually show this - rather than being in a generic message/location? Just checking I have that right.

Reply:
Exactly! Client side code [over here](https://github.com/open-policy-agent/vscode-opa/compare/main...anderseknert:vscode-opa:lsp-eval).